### ISSUE TITLE
make it run faster

### ISSUE THREAD

**Issue by ejrgilbert** [ejrgilbert/whamm#23]
I'm _guessing_ that it's running slowly with the behavior tree because of having to clone the probe each time it processes a new instruction, try seeing if it's possible to remove the clone, or at least, just clone the predicate.

**Comment by ahuoguo** [comment]
Some docs on the flame graph.
```
> cargo install --locked samply
> samply record whamm instr --script tests/scripts
/fault_injection/dfinity/dfinity_async_strcmp_fn.mm --app tests/apps/dfinity/users.wasm
```
![image](https://github.com/ejrgilbert/whamm/assets/52595524/61df8da4-8bfa-4d46-9427-51e4312f4642)
94% of the time is spent on the two regex functions in [`define_compiler_var`](https://github.com/ejrgilbert/whamm/blob/71f737cd28c5d27cc2738de4a12c0e2b144ccd30/src/generator/emitters.rs#L1578-L1584). My guess is the bad time complexity of doing regex matches for the entire wasm file (since it's worst-case exponential as I recall people doing DoS attacks with regex matching).

**Comment by ejrgilbert** [comment]
Fixed in #83
